sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample4_StronglyTypingARecognizedForm.md

```diff
@@ -0,0 +1,79 @@
+string endpoint = "<endpoint>";
+string apiKey = "<apiKey>";
+var credential = new AzureKeyCredential(apiKey);
+var client = new FormRecognizerClient(new Uri(endpoint), credential);
+```
+
+## Strongly-typing a receipt
+
+`FormField<T>` is a helper class used in this sample as a strongly-typed version of `FormField`. They have the same properties, except for `Value`, which returns a `T` instead of a `FieldValue`.
+
+The known receipt fields returned by the service, such as `MerchantName` and `TransactionDate`, will be strongly-typed and put together into another helper class called `Receipt`.
+
+```C# Snippet:FormRecognizerSampleStronglyTypingARecognizedForm
+RecognizedFormCollection recognizedForms = await client.StartRecognizeReceiptsFromUri(new Uri(receiptUri)).WaitForCompletionAsync();
+
+foreach (RecognizedForm recognizedForm in recognizedForms)
+{
+    Receipt receipt = new Receipt(recognizedForm);
+
+    if (receipt.MerchantName != null)
+    {
+        string merchantName = receipt.MerchantName;
+        Console.WriteLine($"Merchant Name: '{merchantName}', with confidence {receipt.MerchantName.Confidence}");
+    }
+
+    if (receipt.TransactionDate != null)
+    {
+        DateTime transactionDate = receipt.TransactionDate;
+        Console.WriteLine($"Transaction Date: '{transactionDate}', with confidence {receipt.TransactionDate.Confidence}");
+    }
+
+    foreach (ReceiptItem item in receipt.Items)
+    {
+        Console.WriteLine("Item:");
+
+        if (item.Name != null)
+        {
+            string name = item.Name;
+            Console.WriteLine($"    Name: '{name}', with confidence {item.Name.Confidence}");
+        }
+
+        if (item.TotalPrice != null)
+        {
+            float totalPrice = item.TotalPrice;
+            Console.WriteLine($"    Total Price: '{totalPrice}', with confidence {item.TotalPrice.Confidence}");
+        }
+    }
+
+    if (receipt.Total != null)
+    {
+        float total = receipt.Total;
+        Console.WriteLine($"Total: '{total}', with confidence {receipt.Total.Confidence}");
+    }
+}
+```
+
+Using `FormField<T>` to make your fields strongly-typed, and populating a custom helper model class, such as `Receipt`, is the recommended approach for handling recognized forms in which expected labels are known.
+
+To see the full example source files, see:
+
+* [Strongly typing a recognized form](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample4_StronglyTypingARecognizedForm.cs)
+* [Receipt class](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Receipt.cs)
+* [ReceiptItem class](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/ReceiptItem.cs)
+* [FormField<T> class](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/FormField{T}.cs)
```

Comment:
I was actually thinking we'd put `FormField<T>` back into the main library to encourage this type of 3rd party extensibility.  What do you think?

Reply:
I was on the fence about this but forgot to mention it. My only concern was that we cannot introduce breaking changes in a type that's in the main library. However, since we are using it as a suggested approach, I think that keeping it stable may be the desired approach.

Reply:
I'm wondering if we can move the `Convert<Type>Field` methods to the main library as well, since they're not likely to be customized by developers, but I don't know where to put them. I thought about having them as public static methods in `FormField<T>`:
```cs
public static FormField<string> ConvertStringField(string fieldName, IReadOnlyDictionary<string, FormField> fields){}
```
Also, if they're public, I don't like the idea of passing the whole dictionary to extract a single field, but I know we're doing this to encapsulate the `TryGetValue` call. Do you think it's better to keep them in `Receipt` for now?

Reply:
I'm not familiar on how we manage extension "things" in our libraries. 
- How much are we expecting for users to use `FormField<T>`? 
- Do they go in a separate namespace? or a different package?

Reply:
- How much are we expecting for users to use `FormField<T>`?
  A lot. This is approach we are suggesting users to always follow when they have known fields in their recognized forms.

- Do they go in a separate namespace? or a different package?
  If we decide to keep `FormField<T>` in our main code, I intended to put it in the main library in the `Models` namespace.